### ISSUE TITLE
Fatal error and high CPU usage

### ISSUE THREAD

**Issue by psydvl** [bash-lsp/bash-language-server#630]
## Details
I'm trying to run LSP on `dotfiles` project in `$HOME` with `.gitignore` with `*`

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

### LSP log panel:

```
LSP-bash: 
LSP-bash: <--- Last few GCs --->
LSP-bash: 
LSP-bash: [26939:0x55b5d80c61c0]   217841 ms: Scavenge 2001.3 (2080.2) -> 1990.7 (2080.4) MB, 14.1 / 0.0 ms  (average mu = 0.252, current mu = 0.252) task;
LSP-bash: [26939:0x55b5d80c61c0]   222013 ms: Mark-sweep 2002.6 (2080.4) -> 1990.1 (2080.9) MB, 4095.7 / 0.0 ms  (average mu = 0.197, current mu = 0.136) task; scavenge might not succeed
LSP-bash: [26939:0x55b5d80c61c0]   226701 ms: Mark-sweep 2002.9 (2080.9) -> 1992.1 (2083.9) MB, 4637.5 / 0.0 ms  (average mu = 0.107, current mu = 0.011) task; scavenge might not succeed
LSP-bash: 
LSP-bash: 
LSP-bash: <--- JS stacktrace --->
LSP-bash: 
LSP-bash: FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
LSP-bash:  1: 0x55b5d540f580 node::Abort() [/usr/bin/node19]
LSP-bash:  2: 0x55b5d5313c49  [/usr/bin/node19]
LSP-bash:  3: 0x55b5d55e7640 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [/usr/bin/node19]
LSP-bash:  4: 0x55b5d55e7a20 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [/usr/bin/node19]
LSP-bash:  5: 0x55b5d57d9217  [/usr/bin/node19]
LSP-bash:  6: 0x55b5d57d930a  [/usr/bin/node19]
LSP-bash:  7: 0x55b5d57e6f9a v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::internal::GarbageCollectionReason, char const*, v8::GCCallbackFlags) [/usr/bin/node19]
LSP-bash:  8: 0x55b5d57e7d50 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/bin/node19]
LSP-bash:  9: 0x55b5d57c5b3f v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/usr/bin/node19]
LSP-bash: 10: 0x55b5d57c6c14 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/usr/bin/node19]
LSP-bash: 11: 0x55b5d57a95ce v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/usr/bin/node19]
LSP-bash: 12: 0x55b5d5b6c0ee v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [/usr/bin/node19]
LSP-bash: 13: 0x55b5d5fb52f9  [/usr/bin/node19]
```

### ps

```sh
$ ps aux | grep --extended-regexp "(LSP-bash|USER)"
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
nixi     29546  145  1.7 11842980 135100 ?     Rl   23:56   0:06 /usr/bin/node19 /home/nixi/.cache/sublime-text/Package Storage/LSP-bash/19.1.0/language-server/node_modules/bash-language-server/bin/main.js start
```

## Addition
For now, I can't set `backgroundAnalysisMaxFiles` due [mistakes](https://github.com/sublimelsp/LSP-bash/issues/43) in LSP-bash

## System info 
bash-language-server version: Version is 4.1.2
node version: v19.1.0
plugin: [LSP-bash](https://github.com/sublimelsp/LSP-bash) 1.1.15
Editor: Sublime Text Build 4143
system: openSUSE Tumbleweed 20221212-0 x86_64

**Comment by skovhus** [comment]
> For now, I can't set backgroundAnalysisMaxFiles due https://github.com/sublimelsp/LSP-bash/issues/43 in LSP-bash

So this turned out to be a sublime plug-in issue, that has been fixed it seems. Did you try configuring backgroundAnalysisMaxFiles?

**Comment by skovhus** [comment]
Is the high CPU usage only when the extension starts, while typing or also when inactive?

Can you share some output from the extension (in vscode, this is done by View -> Output -> Bash IDE) while this is ongoing? 

And does it change anything if you open a small workspace with few files compared to opening a large workspace (e.g. home directory)?

**Comment by skovhus** [comment]
Are you using ShellCheck? 

A new version bash-language-server@4.2.4 has just been released, introducing a short execution delay and debouncing for ShellCheck tasks. 

Let me know if the new version works for you. This is also out in the latest vscode extension.

**Comment by skovhus** [comment]
Note that we have [fixed a bug causing memory leaks and high CPU usage for workspaces with many files](https://github.com/bash-lsp/bash-language-server/pull/661
) – this has been released as bash-language-server@4.2.5 and the vscode extension 1.26.0.

**Comment by psydvl** [comment]
Sorry for long delay
Yes, problem solved in 4.2.5
Thank you
